cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1361,6 +1361,48 @@ gdv_int32 ascii_utf8(const char* data, gdv_int32 data_len) {
   return static_cast<gdv_int32>(data[0]);
 }
 
+// Returns the ASCII character having the binary equivalent to A.
+// If A is larger than 256 the result is equivalent to chr(A % 256).
+FORCE_INLINE
+const char* chr_int64(gdv_int64 context, gdv_int64 in, gdv_int32* out_len) {
+  if (in < 0) {
```

Comment:
Even for negative numbers, chr is returning valid values

For e.g. chr(-158) = 'b'

Reply:
Change the behavior to handle negative number as well